### PR TITLE
feat: add signAllTransactions and signAndSendTransaction

### PR DESCRIPTION
**Pull request by zoruka**
# Description

This PR adds:
- New solana methods `signAllTransactions` and `signAndSendTransactions`
- Verification if wallet supports method

### CODE REVIEW

File: packages/walletconnect-solana/src/core.ts [hunk 1]
```diff
@@ -233,4 +277,22 @@ export class WalletConnectWallet {
...
+	private serialize(transaction: Transaction | VersionedTransaction): string {
+		return Buffer.from(transaction.serialize({ verifySignatures: false })).toString('base64')
+	}
+
+	private deserialize(serializedTransaction: string, versioned = false): Transaction | VersionedTransaction {
+		if (versioned) {
+			return VersionedTransaction.deserialize(Buffer.from(serializedTransaction, 'base64'))
+		}
+
+		return Transaction.from(Buffer.from(serializedTransaction, 'base64'))
```
Comment (tomiir) [inline packages/walletconnect-solana/src/core.ts:290]: same

File: packages/walletconnect-solana/src/core.ts [hunk 2]
```diff
@@ -148,67 +158,101 @@ export class WalletConnectWallet {
...
+				method: WalletConnectRPCMethods.signAndSendTransaction,
+				params: { transaction: this.serialize(transaction) },
+			},
+		})
+
+		return signature
+	}
+
+	async signAllTransactions<T extends Transaction | VersionedTransaction>(transactions: T[]): Promise<T[]> {
+		this.checkIfWalletSupportsMethod(WalletConnectRPCMethods.signAllTransactions)
```
Comment (glitch-txs) [inline packages/walletconnect-solana/src/core.ts:234]: how can the dapp know before hand if the wallet supports these methods?

Comment (zoruka) [reply packages/walletconnect-solana/src/core.ts:234]: It is present in the namespace we receive when there is a successful connection. We can see [here](https://github.com/WalletConnect/walletconnect-solana-adapter/pull/42/files#diff-4ef08e8776fc9a3ab06dd2bb2c371ec838f9e6c76b07f54eb426b73f8935f44cR294) the logic.

Comment (glitch-txs) [reply packages/walletconnect-solana/src/core.ts:234]: I mean the dev using the adapter

Comment (zoruka) [reply packages/walletconnect-solana/src/core.ts:234]: I'm not sure if I understood the question, but he could get it in the session or catch the error

**Comment by kieranbop1990** [comment]
How far off is this being merged? Thanks

**Review by glitch-txs — APPROVED** [review]

**Pull request merged**